feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go

```diff
@@ -0,0 +1,1946 @@
+			WithID(18).AddNextHop(1000, 1),
+
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(1001).WithDecapsulateHeader(fluent.IPinIP).WithEncapsulateHeader(fluent.IPinIP).
+			WithIPinIP(ipv4OuterSrc222, gribiIPv4EntryVRF2222).
+			WithNextHopNetworkInstance(niTeVrf222),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithID(19).AddNextHop(1001, 1),
+	)
+	if err := awaitTimeout(args.ctx, t, args.client, time.Minute); err != nil {
+		t.Logf("Could not program entries via client, got err, check error codes: %v", err)
+	}
+
+	// Programming AFT entries for prefixes in TE_VRF_111
+	args.client.Modify().AddEntry(t,
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(1).WithIPAddress(gribiIPv4EntryDefVRF1),
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(2).WithIPAddress(gribiIPv4EntryDefVRF2),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithID(1).AddNextHop(1, 1).AddNextHop(2, 3).WithBackupNHG(18),
+		fluent.IPv4Entry().WithNetworkInstance(niTeVrf111).
+			WithPrefix(gribiIPv4EntryVRF1111+"/"+maskLen32).WithNextHopGroup(1),
+
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(3).WithIPAddress(gribiIPv4EntryDefVRF4),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithID(4).AddNextHop(3, 2).WithBackupNHG(19),
+		fluent.IPv4Entry().WithNetworkInstance(niTeVrf111).
+			WithPrefix(gribiIPv4EntryVRF1112+"/"+maskLen32).WithNextHopGroup(4),
+	)
+	if err := awaitTimeout(args.ctx, t, args.client, time.Minute); err != nil {
+		t.Logf("Could not program entries via client, got err, check error codes: %v", err)
+	}
+
+	teVRF111IPList := []string{gribiIPv4EntryVRF1111, gribiIPv4EntryVRF1112}
+	for ip := range teVRF111IPList {
+		chk.HasResult(t, args.client.Results(t),
+			fluent.OperationResult().
+				WithIPv4Operation(teVRF111IPList[ip]+"/32").
+				WithOperationType(constants.Add).
+				WithProgrammingResult(fluent.InstalledInFIB).
+				AsResult(),
+			chk.IgnoreOperationID(),
+		)
+	}
+
+	// Programming AFT entries for prefixes in ENCAP_TE_VRF_A
+	args.client.Modify().AddEntry(t,
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(201).WithEncapsulateHeader(fluent.IPinIP).
+			WithIPinIP(ipv4OuterSrc111, gribiIPv4EntryVRF1111).
+			WithNextHopNetworkInstance(niTeVrf111),
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(202).WithEncapsulateHeader(fluent.IPinIP).
+			WithIPinIP(ipv4OuterSrc111, gribiIPv4EntryVRF1112).
+			WithNextHopNetworkInstance(niTeVrf111),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithID(11).AddNextHop(201, 1).AddNextHop(202, 3).WithBackupNHG(18),
```

Comment:
Here for the prefix in Encap VRF are supposed to have below Backup NHG as per readme ?

IPv4Entry {138.0.11.0/24 (ENCAP_TE_VRF_A)} -> NHG#101 (DEFAULT VRF) -> {
  {NH#101, DEFAULT VRF, weight:1},
  {NH#102, DEFAULT VRF, weight:3},
  **backup_next_hop_group: 200 // in case specific vendor implementation or bugs pruned the NHs.**
}

NHG#200 (Default VRF) {
  {NH#200, DEFAULT VRF, weight:1}
}
NH#200 -> {
    network_instance: "DEFAULT"
}